### ISSUE TITLE
Customer using userlocal maui install unable to find maui workloads

### ISSUE THREAD

**Issue by marcpopMSFT** [dotnet/sdk#44165]
Copied the text from here for @anpin I felt we should have a new issue tracking this since the original userlocal issue was addressed as far as I know (the code is in the RC2 branch and we have confirmation of it working).
https://github.com/dotnet/sdk/issues/37706#issuecomment-2408549541
https://github.com/dotnet/sdk/pull/41735#issuecomment-2406380139

For me RC2 does not fix it. Exactly the same as before. 
```console
> dotnet --info
.NET SDK:
 Version:           9.0.100-rc.2.24474.11
 Commit:            315e1305db
 Workload version:  9.0.100-manifests.4872d5d5
 MSBuild version:   17.12.0-preview-24473-03+fea15fbd1

Runtime Environment:
 OS Name:     nixos
 OS Version:  24.11
 OS Platform: Linux
 RID:         linux-x64
 Base Path:   /nix/store/ckjmw8ahlv89vhn76rdqb04r7xkngz9c-dotnet-sdk-9.0.100-rc.2.24474.11/sdk/9.0.100-rc.2.24474.11/

.NET workloads installed:
 [maui-android]
   Installation Source: SDK 9.0.100-rc.2
   Manifest Version:    9.0.0-rc.2.24503.2/9.0.100-rc.2
   Manifest Path:       /home/a/.dotnet/sdk-manifests/9.0.100-rc.2/microsoft.net.sdk.maui/9.0.0-rc.2.24503.2/WorkloadManifest.json
   Install Type:        FileBased

Configured to use loose manifests when installing new manifests.

Host:
  Version:      9.0.0-rc.2.24473.5
  Architecture: x64
  Commit:       990ebf52fc

.NET SDKs installed:
  9.0.100-rc.2.24474.11 [/nix/store/fa8svf3jpqs5jpx9jmigkvcpc7vflyqx-dotnet-core-combined/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 9.0.0-rc.2.24474.3 [/nix/store/fa8svf3jpqs5jpx9jmigkvcpc7vflyqx-dotnet-core-combined/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 9.0.0-rc.2.24473.5 [/nix/store/fa8svf3jpqs5jpx9jmigkvcpc7vflyqx-dotnet-core-combined/shared/Microsoft.NETCore.App]

Other architectures found:
  None

Environment variables:
  DOTNET_ROOT       [/nix/store/fa8svf3jpqs5jpx9jmigkvcpc7vflyqx-dotnet-core-combined]

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download
> dotnet workload list

Installed Workload Id      Manifest Version                     Installation Source
-----------------------------------------------------------------------------------
maui-android               9.0.0-rc.2.24503.2/9.0.100-rc.2      SDK 9.0.100-rc.2

Use `dotnet workload search` to find additional workloads to install.

> dotnet build
  Determining projects to restore...
/nix/store/ckjmw8ahlv89vhn76rdqb04r7xkngz9c-dotnet-sdk-9.0.100-rc.2.24474.11/sdk/9.0.100-rc.2.24474.11/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.ImportWorkloads.targets(38,5): error NETSDK1147: To build this project, the following workloads must be installed: maui-android [/home/a/projects/maui-on-nix/src/MauiOnNix.csproj::TargetFramework=net8.0-android]
/nix/store/ckjmw8ahlv89vhn76rdqb04r7xkngz9c-dotnet-sdk-9.0.100-rc.2.24474.11/sdk/9.0.100-rc.2.24474.11/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.ImportWorkloads.targets(38,5): error NETSDK1147: To install these workloads, run the following command: dotnet workload restore [/home/a/projects/maui-on-nix/src/MauiOnNix.csproj::TargetFramework=net8.0-android]

Build FAILED.

/nix/store/ckjmw8ahlv89vhn76rdqb04r7xkngz9c-dotnet-sdk-9.0.100-rc.2.24474.11/sdk/9.0.100-rc.2.24474.11/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.ImportWorkloads.targets(38,5): error NETSDK1147: To build this project, the following workloads must be installed: maui-android [/home/a/projects/maui-on-nix/src/MauiOnNix.csproj::TargetFramework=net8.0-android]
/nix/store/ckjmw8ahlv89vhn76rdqb04r7xkngz9c-dotnet-sdk-9.0.100-rc.2.24474.11/sdk/9.0.100-rc.2.24474.11/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.ImportWorkloads.targets(38,5): error NETSDK1147: To install these workloads, run the following command: dotnet workload restore [/home/a/projects/maui-on-nix/src/MauiOnNix.csproj::TargetFramework=net8.0-android]
    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:00.76
> ls -al $DOTNET_ROOT/metadata/workloads/9.0.100/userlocal
.r--r--r-- 0 root 31 Dec  1969 /nix/store/fa8svf3jpqs5jpx9jmigkvcpc7vflyqx-dotnet-core-combined/metadata/workloads/9.0.100/userlocal
```

**Comment by marcpopMSFT** [comment]
@anpin can you confirm that the workload manifests for maui are installed in the sdk-manifests folder and that there are maui packs installed in the dotnet\packs folder?

I'm wondering if the packs are installed in one location but we're looking for them in a different location. @tmds @dsplaisted is there an environment variable that controls the location of the userlocal packs lookup? I recall there was.

**Comment by tmds** [comment]
@anpin can you check if having two `userlocal` files as described in https://github.com/dotnet/sdk/issues/37706#issuecomment-2137572573 makes the command work?

**Comment by tmds** [comment]
@marcpopMSFT @dsplaisted The issue is specific to the NixOS packaging. The .NET root is split across different directories which are then combined through symlinks. Due do this split, the SDK ends up looking for the `userlocal` file at the wrong place.

@anpin will someone be addressing this in the NixOS packaging?

**Comment by anpin** [comment]
working nixos shell for dotnet-android / maui  

https://github.com/anpin/maui-on-nix/blob/7a5a1162dd69db2a7c206b243d288d03819d8715/shell.nix

**Comment by marcpopMSFT** [comment]
Sounds like you got workloads unblocked. Thanks for the help on this one tmds
